OrbitCore/SamplingProfiler.cpp

```diff
@@ -423,8 +424,8 @@ void SamplingProfiler::AddAddress(uint64_t a_Address) {
 
     m_ExactAddressToFunctionAddress[a_Address] =
         symbol_info->Address ? symbol_info->Address : a_Address;
-    m_AddressToName[a_Address] = symName;
-    m_AddressToName[symbol_info->Address] = symName;
+    Capture::GAddressToFunctionName[a_Address] = symName;
```

Comment:
It seems this global is added so that we can add it to the serialized capture data.  Could we serialize the entire SamplingProfiler object instead of breaking it down and adding global variables?

Reply:
Actually I don't understand why it was made global, it used to be serialized as part of the SamplingProfiler already.

Reply:
Hopefully I remember correctly: In `TimeGraph::SelectEvents` when a new selection is made a **new** `SamplingProfiler` is created. These obviously cannot be serialized as they are different for each selection. On such `SamplingProfiler`s, `SamplingProfiler::ProcessSamples` is called, and it turns `SamplingProfiler::ResolveCallstacks`, which repeatedly calls `AddAddress`, which used to fill `m_AddressToName`. Some of these names could come from `Process::GetFunctionFromAddress` and `Process::GetLinuxAddressInfo`, but the process might not be there anymore, and I chose not to load `Capture::GTargetProcess` as it causes a lot of inconsistencies in the UI (`Capture::GTargetProcess` is currently used as both the selected process and the captured process, but should only be the former).
This would basically cause all selections to not have function names.
That's why I made this global, so that all `SamplingProfiler`s can share it, and so that it can be saved and loaded independently of both `SamplingProfiler` and `Process`. Ideally, it would be part of a `Capture` class.

Reply:
Ok, it seems there is some clean up that could be done here but that would be beyond the scope of this PR.  Let's keep it as is for now, and eventually revisit.  Thanks.